encoder.go

```diff
@@ -117,9 +117,9 @@ func EncodeFloat64(w io.Writer, f float64) (err error) {
 
 const (
 	// seconds of 01-01-0001
-	minSeconds = -62135596800
+	minSeconds = int64(-62135596800)
```

Comment:
you can also do

```suggestion
	minSeconds int64 = -62135596800
```

, which I prefer slightly better